src/Dumplie/Application/Command/Customer/CreateCartHandler.php

```diff
@@ -38,11 +38,10 @@ public function __construct(Carts $carts, Factory $factory)
      */
     public function handle(CreateCart $command)
     {
-        $this->carts->add(new Cart(new CartId($command->uuid()), $command->currency()));
-
         $transaction = $this->factory->open();
 
         try {
+            $this->carts->add(new Cart(new CartId($command->uuid()), $command->currency()));
```

Comment:
Geberating uuid in UI give you possibility to fetch created resource without returning anything from handler. It's easiest way to get access to data created by command.